Observer/Category.php

```diff
@@ -0,0 +1,80 @@
+     * @param \Metrilo\Analytics\Helper\ApiClient                $apiClient
+     * @param \Metrilo\Analytics\Helper\CustomerSerializer       $customerSerializer
+     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(
+        \Metrilo\Analytics\Helper\Data                                  $helper,
+        \Metrilo\Analytics\Helper\ApiClient                             $apiClient,
+        \Metrilo\Analytics\Helper\CategorySerializer                    $categorySerializer,
+        \Magento\Catalog\Model\ResourceModel\Category\CollectionFactory $categoryCollection,
+        \Magento\Framework\App\Config\ScopeConfigInterface              $scopeConfig,
+        \Magento\Store\Model\StoreManagerInterface                      $storeManager
+    ) {
+        $this->helper             = $helper;
+        $this->apiClient          = $apiClient;
+        $this->categorySerializer = $categorySerializer;
+        $this->categoryCollection = $categoryCollection;
+        $this->scopeConfig        = $scopeConfig;
+        $this->storeManager       = $storeManager;
+    }
+    
+    private function getStoreIdPerProject($storeIds) {
+        $storeIdConfigMap = [];
+        foreach ($storeIds as $storeId) {
+            if ($storeId == 0) { // store 0 is always admin
+                continue;
+            }
+            $storeIdConfigMap[$storeId] = $this->scopeConfig
+                                               ->getValue(
+                                                 'metrilo_analytics/general/api_key',
+                                                 \Magento\Store\Model\ScopeInterface::SCOPE_STORE,
+                                                  $storeId
+                                               );
+        }
+        $storeIdConfigMap = array_unique($storeIdConfigMap);
+        
+        return array_keys($storeIdConfigMap);
+    }
+    
+    private function getCategoryObjectWithRequestPath($categoryId, $storeId) {
+        $this->storeManager->setCurrentStore($storeId);
+        
+        return $this->categoryCollection
+                    ->create()
+                    ->addAttributeToSelect('name')
+                    ->addAttributeToFilter('entity_id', $categoryId)
+                    ->addUrlRewriteToResult()
+                    ->getFirstItem();
+    }
+    
+    public function execute(Observer $observer)
+    {
+        try {
+            $category = $observer->getEvent()->getCategory();
+            if ($category->getStoreId() == 0) {
+                $categoryStoreIds = $this->getStoreIdPerProject($category->getStoreIds());
+            } else {
+                $categoryStoreIds[] = $category->getStoreId();
+            }
+            foreach ($categoryStoreIds as $storeId) {
+                $categoryObject = $this->getCategoryObjectWithRequestPath($category->getId(), $storeId);
+                $categoryObject->setStoreId($storeId);
```

Comment:
I think it makes more sense move this to the `getCategoryObjectWithRequestPath` method, since it is part of the object setup.